code/lib/csf-tools/src/CsfFile.ts

```diff
@@ -200,6 +202,19 @@ export class CsfFile {
         } else if (['includeStories', 'excludeStories'].includes(p.key.name)) {
           (meta as any)[p.key.name] = parseIncludeExclude(p.value);
         } else if (p.key.name === 'component') {
+          let n = p.value;
+          if (t.isIdentifier(n)) {
+            const id = n.name;
+            const importStmt = program.body.find((stmt) => (
+              t.isImportDeclaration(stmt) && stmt.specifiers.find((spec) => spec.local.name === id)
+            )) as t.ImportDeclaration;
+            if (importStmt) {
+              const { source } = importStmt;
+              if (t.isStringLiteral(source)) {
+                this._componentPath = source.value;
```

Comment:
Amazing! <3
Will this end up getting the file extension as well? for instance:
```ts
import { Button } from './MyButton'
```

How will it know that the path would be e.g. `src/components/Button.tsx`? I know the `importPath` property does have the extension, for the story

Reply:
Yeah it probably needs to be enhanced to figure out whether it's a file or a package and to fix the file extension and relative paths. I'll try to handle that in the indexer.